### PR TITLE
POZ: catering for multiple ffdc packets for POZ chips

### PR DESCRIPTION
**Pull request by devenrao**

### CODE REVIEW

**Review by deepakala-k — APPROVED** [review]

**Review by Swarnendu-R-C — APPROVED** [review]

File: dump/dump_collect.cpp [hunk 1]
```diff
@@ -298,12 +316,11 @@ void collectDump(const uint8_t type, const uint32_t id,
             struct pdbg_target* ocmbTarget;
             pdbg_for_each_target("ocmb", target, ocmbTarget)
             {
-                if (pdbg_target_probe(ocmbTarget) != PDBG_TARGET_ENABLED)
+                if (!is_ody_ocmb_chip(ocmbTarget))
```
Comment (dhruvibm) [inline dump/dump_collect.cpp:319]: Is this relevant to the current scope?

File: dump/dump_collect.cpp [hunk 2]
```diff
@@ -95,7 +96,7 @@ void collectDumpFromSBE(struct pdbg_target* chip,
 {
     using namespace phosphor::logging;
     auto chipPos = pdbg_target_index(chip);
-    bool isOcmb = openpower::phal::sbe::is_ody_ocmb_chip(chip);
+    bool isOcmb = is_ody_ocmb_chip(chip);
```
Comment (dhruvibm) [inline dump/dump_collect.cpp:99]: please limit changes only for the current scope

File: dump/dump_collect.cpp [hunk 3]
```diff
@@ -141,21 +142,38 @@ void collectDumpFromSBE(struct pdbg_target* chip,
                 type, clockState, chipPos, collectFastArray, sbeError.what())
                 .c_str());
 
-        std::string event = "org.open_power.Processor.Error.SbeChipOpFailure";
         auto dumpIsRequired = false;
-
-        if (sbeError.errType() == openpower::phal::exception::SBE_CMD_TIMEOUT)
+        std::string event;
+        if (isOcmb)
```
Comment (dhruvibm) [inline dump/dump_collect.cpp:147]: Now SBE_CMD_TIMEOUT will come for ocmb also, so if we make changes here we will have to make all changes to collect SBE dump. which was not done because the SBE dump was not ready when doing it.

I think we can do this as a separate pull request.
for this commit move line 196 above 180

**Review by dhruvibm — CHANGES_REQUESTED** [review]

File: dump/dump_collect.cpp (hunk 1 above)
Comment (devenrao) [reply dump/dump_collect.cpp:319]: i put it in different commit as i find this logic wrong, agree should have been different PR

File: dump/dump_collect.cpp (hunk 2 above)
Comment (devenrao) [reply dump/dump_collect.cpp:99]: ok

File: dump/create_pel.cpp [hunk 4]
```diff
@@ -34,46 +34,29 @@ constexpr auto loggingObjectPath = "/xyz/openbmc_project/logging";
 constexpr auto loggingInterface = "xyz.openbmc_project.Logging.Create";
 constexpr auto opLoggingInterface = "org.open_power.Logging.PEL";
 
-uint32_t createSbeErrorPEL(const std::string& event, const sbeError_t& sbeError,
-                           const FFDCData& ffdcData, const Severity& severity)
+using FFDCInfo = std::vector<std::tuple<
+    sdbusplus::xyz::openbmc_project::Logging::server::Create::FFDCFormat,
+    uint8_t, uint8_t, sdbusplus::message::unix_fd>>;
+
+uint32_t createPelWithFFDCfiles(const std::string& event,
```
[the remark this answers is not in the document]
Comment (devenrao) [reply dump/create_pel.cpp:41]: new helper method to put common code for proc sbe error and poz sbe error pel

File: dump/create_pel.cpp [hunk 5]
```diff
@@ -34,46 +34,29 @@ constexpr auto loggingObjectPath = "/xyz/openbmc_project/logging";
...
-
-    std::vector<std::tuple<
-        sdbusplus::xyz::openbmc_project::Logging::server::Create::FFDCFormat,
-        uint8_t, uint8_t, sdbusplus::message::unix_fd>>
-        pelFFDCInfo;
+        auto bus = sdbusplus::bus::new_default();
 
-    // get SBE ffdc file descriptor
-    auto fd = sbeError.getFd();
+        std::unordered_map<std::string, std::string> additionalData;
```
[the remark this answers is not in the document]
Comment (devenrao) [reply dump/create_pel.cpp:52]: if you look left side line 41 to 32 we are declaring a map and  initializing the map with _PID and SBE_ERR_MSG and agan we are filling it with _PID and SBE_ERR_MSG at line 45 and 46 which I had fixed from lin 52 to 54 right hand side.

line 48 to 51 from left hand side came to line 55 to 58 right hand side

File: dump/dump_collect.cpp (hunk 3 above)
Comment (devenrao) [reply dump/dump_collect.cpp:147]: done as discussed

**Review by dhruvibm — APPROVED** [review]
